src/operator/tensor/indexing_op.cc

```diff
@@ -129,7 +186,7 @@ NNVM_REGISTER_OP(one_hot)
 .set_attr<nnvm::FInferType>("FInferType", OneHotOpType)
 .set_attr<FCompute>("FCompute<cpu>", OneHotOpForward<cpu>)
 .set_attr<nnvm::FGradient>("FGradient", MakeZeroGradNodes)
-.add_argument("indices", "NDArray", "array of locations where to set on_value")
+.add_argument("indices", "ndarray-or-symbol", "array of locations where to set on_value")
```

Comment:
https://github.com/dmlc/mxnet/blob/master/python/mxnet/_ctypes/ndarray.py#L76

Changing it to NDArray-or-Symbol should make it backward compatible